### DISCUSSION TITLE
About mathematical funtions in Penrose

### DISCUSSION THREAD

**Discussion by HZT3434** [penrose/penrose#949] [Ideas]
Is it possible to draw special lines with specific functions? Like y=ax?

**Comment by samestep** [comment]
Hello @HZT3434! Sorry for taking so long to respond; I put together this gist a few days ago and was going to post it here but forgot: https://gist.github.com/samestep/4344820d0cafe2c69195cfa9cb05def9

If you put this in https://penrose.cs.cmu.edu/try/ it should draw the graph of the function _y_=_ax_ for _a_=0.5; is this what you intended? The specific value of _a_ is specified in Style; did you instead mean to put the number itself in Substance?

**Comment by HZT3434** [comment]
Hi！Thank you for the kind response and I will try the new example. 


Regards,
Huang







At 2022-05-14 08:08:04, "Sam Estep" ***@***.***> wrote:

Hello @HZT3434! Sorry for taking so long to respond; I put together this gist a few days ago and was going to post it here but forgot: https://gist.github.com/samestep/4344820d0cafe2c69195cfa9cb05def9

If you put this in https://penrose.cs.cmu.edu/try/ it should draw the graph of the function y=ax for a=0.5; is this what you intended? The specific value of a is specified in Style; did you instead mean to put the number itself in Substance?

**Comment by HZT3434** [comment]
As a mathematics phd I really love the penrose programe language. Now my question is: can penrose do things like matplot? I am expecating the evolutions! Thank you!


Regards,
Huang













At 2022-05-14 08:08:04, "Sam Estep" ***@***.***> wrote:

Hello @HZT3434! Sorry for taking so long to respond; I put together this gist a few days ago and was going to post it here but forgot: https://gist.github.com/samestep/4344820d0cafe2c69195cfa9cb05def9

If you put this in https://penrose.cs.cmu.edu/try/ it should draw the graph of the function y=ax for a=0.5; is this what you intended? The specific value of a is specified in Style; did you instead mean to put the number itself in Substance?

**Reply by samestep** [reply]
Good question! That sort of quantitative plot is more outside the scope of Penrose. We are specifically targeting mathematical/conceptual diagrams, rather than data visualization.

**Comment by HZT3434** [comment]
I see. Still it is great to have penrose to solve the mathematical/conceptual diagrams. Thank you!


Regards,
Huang
















At 2022-05-16 22:11:52, "Sam Estep" ***@***.***> wrote:

Good question! That sort of quantitative plot is more outside the scope of Penrose. We are specifically targeting mathematical/conceptual diagrams, rather than data visualization.
